### ISSUE TITLE
Use `FSNode`'s `Type` method instead of casting it to a `File` or `Directory`

### ISSUE THREAD

**Issue by schomatis** [ipfs/go-mfs#22]
This is actually an issue for the clients of this layer like `go-ipfs`.

Also, helper functions can be added like `isFile()` or `isDir`.

**Comment by overbool** [comment]
@schomatis Did you mean the follow code?

https://github.com/ipfs/go-ipfs/blob/9bf4e4145ea897ced2f9aa2087296b3beda6e941/core/commands/files.go#L455-L458

Should we add `IsFile` and `IsDir`, and then use `fsn.IsFile()` and `fsn.IsDir` instead of direct `switch case` about `mfs.Directory` and `mfs.File`?

**Comment by schomatis** [comment]
Yes! In that particular case it's fine to use the already existing `Type()` and switch on those values but in others like

https://github.com/ipfs/go-ipfs/blob/9bf4e4145ea897ced2f9aa2087296b3beda6e941/core/commands/files.go#L1109

where only one type is being asserted (e.g., `*mfs.Directory`) we can use a (new) helper function like `IsDir`.

But in general what I want to avoid are references to `mfs.Directory` and `mfs.File` which are the internal implementation of the MFS directory and file and use the public API instead.

**Comment by overbool** [comment]
@schomatis Which solution do you think is better? 

1. `fsn.IsDir() /  fsn.IsFile()`(this should add method in FSNode interface and implement by `File` and `Directory`) .
1.  `mfs.IsDir(fsn) / mfs.IsFile(fsn)`

**Comment by schomatis** [comment]
The second one, that way we don't need to modify the `FSNode` interface with helper methods.

**Comment by aschmahmann** [comment]
This repository is no longer maintained and has been copied over to [Boxo](https://github.com/ipfs/boxo).  In an effort to avoid noise and crippling in the Boxo repo from the weight of issues of the past, we are closing most issues and PRs in this repo.  Please feel free to open a new issue in Boxo (and reference this issue) if resolving this issue is still critical for unblocking or improving your usecase.

You can learn more in the [FAQs for the Boxo repo copying/consolidation effort](https://github.com/ipfs/boxo/wiki/Copied-or-Migrated-Repos-FAQ).
